lib/macro.js

```diff
@@ -0,0 +1,157 @@
+var constants = require('./macro-constants');
+
+function Macro(config, macroManager) {
+  var macroIdentity = this;
+  this.callersCallback = null;
+  this.config = config;
+  this.name = config.name;
+  this.macroClosureChain = null;
+
+  function createMacroStep(done, stepDescription) {
```

Comment:
Similar feedback here - it's more conventional to put the `callback` as the final parameter. Also, out of curiosity, is there a reason this is called `done` vs `callback`?

Reply:
'done' differs to 'callback' in the convention that 'done' is allways provided by the caller. One can and should rely on it so that any 'if callback != null then callback()' stuff is necessary. In this context I'd like to stay with 'done' as long as we don't refactor to other things like async await or promises.